src/libraries/System.Security.Cryptography.Cose/src/System/Security/Cryptography/Cose/CoseHeaderLabel.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+
+namespace System.Security.Cryptography.Cose
+{
+    [DebuggerDisplay("Label = {LabelAsInt32.HasValue ? LabelAsInt32 : LabelAsString}, Type = {LabelAsInt32.HasValue ? typeof(int) : typeof(string),nq}")]
+    public readonly struct CoseHeaderLabel : IEquatable<CoseHeaderLabel>
+    {
+        // https://www.iana.org/assignments/cose/cose.xhtml#header-parameters
+        public static readonly CoseHeaderLabel Algorithm = new CoseHeaderLabel(KnownHeaders.Alg);
+        public static readonly CoseHeaderLabel Critical = new CoseHeaderLabel(KnownHeaders.Crit);
+        public static readonly CoseHeaderLabel ContentType = new CoseHeaderLabel(KnownHeaders.ContentType);
+        public static readonly CoseHeaderLabel KeyIdentifier = new CoseHeaderLabel(KnownHeaders.Kid);
+        public static readonly CoseHeaderLabel IV = new CoseHeaderLabel(KnownHeaders.IV);
+        public static readonly CoseHeaderLabel PartialIV = new CoseHeaderLabel(KnownHeaders.PartialIV);
+        public static readonly CoseHeaderLabel CounterSignature = new CoseHeaderLabel(KnownHeaders.CounterSignature);
```

Comment:
These should likely be properties rather than fields, e.g.
```suggestion
        public static CoseHeaderLabel Algorithm => new CoseHeaderLabel(KnownHeaders.Alg);
        public static CoseHeaderLabel Critical => new CoseHeaderLabel(KnownHeaders.Crit);
        public static CoseHeaderLabel ContentType => new CoseHeaderLabel(KnownHeaders.ContentType);
        public static CoseHeaderLabel KeyIdentifier => new CoseHeaderLabel(KnownHeaders.Kid);
        public static CoseHeaderLabel IV => new CoseHeaderLabel(KnownHeaders.IV);
        public static CoseHeaderLabel PartialIV => new CoseHeaderLabel(KnownHeaders.PartialIV);
        public static CoseHeaderLabel CounterSignature => new CoseHeaderLabel(KnownHeaders.CounterSignature);
```

I assume we still need to do an API review on the whole surface area?

Reply:
> I assume we still need to do an API review on the whole surface area?

Yeah.  This is doing one of the "start flowing for partner early adoption feedback, and we've told them to expect breaking changes" experiments.  IIRC something somewhere should be painting this with preview-opt-in.